pkg/virt-launcher/virtwrap/network/podinterface_test.go

```diff
@@ -183,11 +189,11 @@ var _ = Describe("Pod Network", func() {
 	TestPodInterfaceIPBinding := func(vm *v1.VirtualMachineInstance, domain *api.Domain) {
 
 		//For Bridge tests
-		mockNetwork.EXPECT().LinkSetName(dummySwap, newPodInterfaceName).Return(nil)
-		mockNetwork.EXPECT().LinkByName(primaryPodInterfaceName).Return(dummy, nil).Times(2)
-		mockNetwork.EXPECT().LinkByName(primaryPodInterfaceName).Return(dummySwap, nil)
+		mockNetwork.EXPECT().LinkSetName(primaryPodInterfaceBeforeNameChange, newPodInterfaceName).Return(nil)
+		mockNetwork.EXPECT().LinkByName(originalPodInterfaceName).Return(primaryPodInterfaceBeforeNameChange, nil)
+		mockNetwork.EXPECT().LinkByName(newPodInterfaceName).Return(primaryPodInterfaceAfterNameChange, nil)
 		mockNetwork.EXPECT().AddrReplace(dummySwap, &fakeAddr).Return(nil)
-		mockNetwork.EXPECT().AddrList(dummy, netlink.FAMILY_V4).Return(addrList, nil)
+		mockNetwork.EXPECT().AddrList(primaryPodInterfaceBeforeNameChange, netlink.FAMILY_V4).Return(addrList, nil)
 		mockNetwork.EXPECT().AddrList(dummy, netlink.FAMILY_ALL).Return(addrList, nil)
```

Comment:
You should use dummy for bridge binding at all.
Here for example you should change to - `mockNetwork.EXPECT().AddrList(primaryPodInterfaceBeforeNameChange, netlink.FAMILY_ALL).Return(addrList, nil)`
Same for lines- 198,200, 203, 204 and 209.
All the occurences of `dummy` and the "//For Bridge tests" block, should be changed to `primaryPodInterfaceBeforeNameChange` (in most of the cases, maybe to `primaryPodInterfaceAfterNameChange` in some cases).
That's why I suggested to change `dummy` to `dummyMasquerade`, so it will be clear in has nothing to do with bridge binding. (Please notice that if you follow this suggestion you will have to create q new variable for dummy in test "should return an error if the MTU is out or range").

Reply:
I'm a little confused.  Are you saying to change `dummy` to `primaryPodInterfaceBeforeNameChange` in `//For Bridge tests` and `dummy` to `dummyMasquerade` everywhere else?  There's also another var called `masqueradeDummy`, should I use that?

Reply:
why are the masquerade tests being lumped in with the bridge binding tests?

Reply:
I agree it is a mess.

>  Are you saying to change dummy to primaryPodInterfaceBeforeNameChange in //For Bridge tests

yes

> and dummy to dummyMasquerade everywhere else? There's also another var called masqueradeDummy, should I use that?

oh no, you're right, we have `masqueradeDummy` which is the extra nic we attach to the bridge in case of masquerade binding. It will be too confisuing to have `dummyMasquerade` as well.
So maybe just remove the `dummy` and use the `primaryPodInterfaceBeforeNameChange` instead in all the places.
I just think `primaryPodInterfaceBeforeNameChange` is missing the MTU property.
`primaryPodInterfaceBeforeNameChange = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: originalPodInterfaceName, MTU: mtu}}`
`

Reply:
You can apply [the patch](https://paste.centos.org/view/5d5a2c68) I created. I fixes the tests of this PR.